src/human-readable/types/signatures.ts

```diff
@@ -1,10 +1,23 @@
-import type { AbiStateMutability } from '../../abi'
-import type { Error } from '../../types'
+import type { AbiStateMutability, AbiType } from '../../abi'
+import type { Error, Trim } from '../../types'
 
 type IsName<T extends string> = T extends '' | `${string}${' '}${string}`
   ? false
   : true
 
+type isValidStructProperty<TProperty extends string> =
+  TProperty extends `${infer Head};${infer Tail}`
+    ? Head extends
+        | `${AbiType}`
+        | `${string} ${string}`
+        | `(${string}) ${string}`
```

Comment:
Think we also need inline tuple without property name:

```suggestion
        | `(${string})
        | `(${string}) ${string}`
```